packages/codemod/src/upgrade-legacy.ts

```diff
@@ -284,37 +284,42 @@ const upgradeLegacy = async () => {
       }
 
       getAllFiles(appDir, [], [], [".ts", ".tsx", ".js", ".jsx"]).forEach((filename) => {
-        const program = getCollectionFromSource(path.resolve(appDir, filename))
-        const parsedProgram = program.get()
-
-        parsedProgram.value.program.body.forEach((e: ImportDeclaration) => {
-          if (e.type === "ImportDeclaration") {
-            if (e.source.value === "blitz") {
-              e.specifiers?.slice().forEach((specifier: any) => {
-                const importedName =
-                  specifier.imported.type === "StringLiteral"
-                    ? specifier.imported.value
-                    : specifier.imported.name
-                if (importedName in specialImports) {
-                  addNamedImport(
-                    program,
-                    importedName,
-                    specialImports[importedName]!,
-                    undefined,
-                    renames[importedName],
-                  )
-                  removeImport(program, importedName, "blitz")
+        try {
+          const program = getCollectionFromSource(path.resolve(appDir, filename))
+          const parsedProgram = program.get()
+
+          parsedProgram.value.program.body.forEach((e: ImportDeclaration) => {
+            if (e.type === "ImportDeclaration") {
+              if (e.source.value === "blitz") {
+                e.specifiers?.slice().forEach((specifier: any) => {
+                  const importedName =
+                    specifier.imported.type === "StringLiteral"
+                      ? specifier.imported.value
+                      : specifier.imported.name
+                  if (importedName in specialImports) {
+                    addNamedImport(
+                      program,
+                      importedName,
+                      specialImports[importedName]!,
+                      undefined,
+                      renames[importedName],
+                    )
+                    removeImport(program, importedName, "blitz")
+                  }
+                })
+                // Removed left over "import 'blitz';"
+                if (!e.specifiers?.length) {
+                  const index = parsedProgram.value.program.body.indexOf(e)
+                  parsedProgram.value.program.body.splice(index, 1)
                 }
-              })
-              // Removed left over "import 'blitz';"
-              if (!e.specifiers?.length) {
-                const index = parsedProgram.value.program.body.indexOf(e)
-                parsedProgram.value.program.body.splice(index, 1)
               }
             }
-          }
-        })
-        fs.writeFileSync(path.resolve(appDir, filename), program.toSource())
+          })
+          fs.writeFileSync(path.resolve(appDir, filename), program.toSource())
+        } catch (e: any) {
```

Comment:
We're being very strict with using any or assertions. You can see how to handle errors here https://github.com/blitz-js/blitz/blob/39fd15d4b5a522740197d1311db16884bd7dfc91/packages/codemod/src/upgrade-legacy.ts#L1301

Reply:
hey, @dillonraphael I have removed all the manual assertions to ```any``` for errors that can be caught in a file. The inner ```try...catch``` will just pass on the error to the outer catch block like you mentioned that should do the check. I was not sure whether to duplicate the logic in all places. If this is what is expected, I shall do that.

Reply:
Yup passing it to the outer catch block should be good @siddhsuresh. But seems the CI is failing.

Reply:
@dillonraphael The CI is now passing.